src/device.rs

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) 2022 Jan Holthuis <jan.holthuis@rub.de>
+//
+// This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy
+// of the MPL was not distributed with this file, You can obtain one at
+// http://mozilla.org/MPL/2.0/.
+//
+// SPDX-License-Identifier: MPL-2.0
+
+//! High-level API for working with Rekordbox device exports.
+
+use crate::{setting, setting::Setting};
+use binrw::BinRead;
+use std::path::Path;
+
+/// Represents a Rekordbox device export.
+#[derive(Debug, Default, PartialEq, Eq, Clone)]
+pub struct DeviceExport {
+    devsetting: Option<Setting>,
+    djmmysetting: Option<Setting>,
+    mysetting: Option<Setting>,
+    mysetting2: Option<Setting>,
+}
+
+impl DeviceExport {
+    fn load_setting(path: &Path) -> Result<Setting, crate::Error> {
+        let mut reader = std::fs::File::open(path)?;
+        let setting = Setting::read(&mut reader)?;
+        Ok(setting)
+    }
+
+    /// Load device export from the given path.
+    ///
+    /// The path should contain a `PIONEER` directory.
+    pub fn load(&mut self, path: &Path) -> Result<(), crate::Error> {
+        let path = path.join("PIONEER");
+        self.devsetting = Some(Self::load_setting(&path.join("DEVSETTING.DAT"))?);
+        self.djmmysetting = Some(Self::load_setting(&path.join("DJMMYSETTING.DAT"))?);
+        self.mysetting = Some(Self::load_setting(&path.join("MYSETTING.DAT"))?);
+        self.mysetting2 = Some(Self::load_setting(&path.join("MYSETTING2.DAT"))?);
+
+        Ok(())
+    }
+
+    /// Get the settings from this export.
+    #[must_use]
+    pub fn get_settings(&self) -> Settings {
+        let mut settings = Settings::default();
+        [
+            &self.mysetting,
+            &self.mysetting2,
+            &self.djmmysetting,
+            &self.devsetting,
+        ]
+        .into_iter()
+        .flatten()
+        .for_each(|setting| match &setting.data {
+            setting::SettingData::MySetting(data) => {
+                settings.set_mysetting(data);
+            }
+            setting::SettingData::MySetting2(data) => {
+                settings.set_mysetting2(data);
+            }
+            setting::SettingData::DJMMySetting(data) => {
+                settings.set_djmmysetting(data);
+            }
+            setting::SettingData::DevSetting(data) => {
+                settings.set_devsetting(data);
+            }
+        });
+
+        settings
+    }
+}
+
+/// Settings object containing for all device settings.
+#[derive(Debug, Default, PartialEq, Eq, Clone)]
+pub struct Settings {
```

Comment:
What do we want the `Default` to be? Just a struct with all `None`s (which is derived) or with the actual default settings from rekordbox?

Reply:
Rekordbox' default settings probably make sense.